lib/sqlalchemy/sql/elements.py

```diff
@@ -430,7 +444,7 @@ def _with_binary_element_type(self, type_):
         return self
 
     @property
-    def _constructor(self):
+    def _constructor(self) -> Any:
```

Comment:
`self.__class__` returns `Type[ClauseElement]`, however this return type is incompatible with `SupportsWrappingAnnotations` supertype - putting `SupportsWrappingAnnotations` in return doesn't help either. Inside `SupportsWrappingAnnotations`, when `TYPE_CHECKING == True`, `_constructor` is an attribute of `Callable[..., SupportsWrappingAnnotations]` type - so even with `Any` return type I get yet another writable attribute override with read-only property.